addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/TestCrossSiteScriptV2.java

```diff
@@ -341,12 +373,52 @@ public void scan(HttpMessage msg, String param, String value) {
                     }
 
                     if (!attackWorked) {
-                        // Try a simple alert attack
+                        for (String scriptAlert : GENERIC_SCRIPT_ALERT_LIST) {
+                            // Try a simple alert attack
+                            List<HtmlContext> contexts2 =
+                                    performAttack(
+                                            msg,
+                                            param,
+                                            context.getSurroundingQuote() + ">" + scriptAlert,
+                                            context,
+                                            HtmlContext.IGNORE_TAG);
+                            if (contexts2 == null) {
+                                return;
```

Comment:
returning instead of break, reason is contexts will only be null incase of Stop event is triggered or URL or some kind of exception occurs. Till now only one loop is there so break behaves sam way. For achieving same behavior returning.